python-modules/cis_publisher/cis_publisher/mozilliansorg_groups.py

```diff
@@ -0,0 +1,122 @@
+import cis_profile
+import cis_publisher
+from cis_publisher.publisher import PublisherError
+import logging
+import json
+from queue import Queue
+from urllib.parse import quote_plus
+
+logger = logging.getLogger(__name__)
+MO_PREFIX = "mozilliansorg_"
+
+
+def get_nested(d, *keys, default=None):
+    if d is None:
+        return default
+    for key in keys:
+        try:
+            d = d[key]
+        except KeyError:
+            return default
+    return d
+
+
+def unpack_string_list(lst=[]):
+    try:
+        return [el["S"][len(MO_PREFIX) :] for el in lst if el["S"].startswith(MO_PREFIX)]
+    except (KeyError, TypeError):
+        logger.error("invalid string list {}".format(lst))
+        return None
+
+
+class MozilliansorgGroupUpdate:
+    def __init__(self, typ, user_id, groups=[]):
+        self.typ = typ
+        self.user_id = user_id
+        self.groups = groups
+
+    def from_record(record):
+        """
+        Constructs a MozilliansGroupUpdate from a raw sqs record.
+        Return a MozilliansGroupUpdate or None to error gracefully.
```

Comment:
Something like this would be good additionally, as per other pieces of code we have (you can look at https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings for inspiration though its not super well defined AFAIK)

```
Arguments:
typ -- it's ... (i dont actually know yet, ive to find the function call and figure out what that is)
user_id -- str the IAM user id
groups -- list of mozilliansorg groups for that user_id
```

This helps understanding what the code does a whole lot without manually tracing back the function calls

note: its a good habit for most functions, though the ones that are really straight forward dont really matter too much (like `unpack_list_from_string` or other small helpers)

same for classes